test_data/rules/unnecessary_late.dart

```diff
@@ -0,0 +1,19 @@
+// Copyright (c) 2021, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+// test w/ `dart test -N unnecessary_late`
+
+late String unnecessaryTopLevel = ""; // LINT
+
+late String necessaryTopLevel; // OK
```

Comment:
Consider also adding a test of the form `String unnecessaryTopLevel = '';` just to test the other possible combination.